### PR TITLE
Use Yarn 3 to manage monorepo installation

### PR DESCRIPTION
**Pull request by laurent22**
Since Lerna is getting slower and slower, and is now almost discontinued, we can try to use Yarn 3 Workspaces instead.

It's way faster on cold install, and extremely fast on warm install. It also seems to handle errors better than npm as it creates a text files about the specific error when something happens (instead of 50 pages of meaningless warnings and errors as is common with npm). Hopefully it can also solve our issues with package-lock files.

- CLI and desktop apps seem to work fine, mobile and server are still to be done
- We can't use Yarn Plug and Play (which would make things even faster) because React Native and Electron don't support it for now
- Most changes are about copying assets outside of the node_modules directories since they no longer exists
- We'll probably still use Lerna for certain tasks, like publishing all the packages in one command, or for "npm run watch" that applies to all the packages. But hopefully we can get rid of the bootstrap command.

# TODO

- [x] Fix clean function so that it doesn't depend on any other package
- [x] Replace calls to npm with yarn

### CODE REVIEW

File: packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx [hunk 1]
```diff
@@ -571,7 +571,8 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 				statusbar: false,
 				target_list: false,
 				table_resize_bars: false,
-				language: ['en_US', 'en_GB'].includes(language) ? undefined : language,
+				language_url: `build/lib/tinymce/langs/${['en_US', 'en_GB'].includes(language)}` ? undefined : language,
```
Comment (laurent22) [inline packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx:574]: Need to fix this - the whole url should be undefined if the language is not supported.

File: packages/app-cli/package.json [hunk 2]
```diff
@@ -34,9 +34,7 @@
     "owner": "Laurent Cozic"
   },
   "version": "2.6.1",
-  "bin": {
-    "joplin": "./main.js"
-  },
+  "bin": "./main.js",
```
Comment (laurent22) [inline packages/app-cli/package.json:37]: Not sure what I've changed here - should check it still works when the app is built.

File: packages/app-desktop/package.json [hunk 3]
```diff
@@ -31,7 +31,8 @@
     "afterSign": "./tools/notarizeMacApp.js",
     "extraResources": [
       "build/icons/*",
-      "build/images/*"
+      "build/images/*",
+      "build/tinymce/langs/*"
```
Comment (laurent22) [inline packages/app-desktop/package.json:35]: Not needed I think.

File: packages/lib/package.json [hunk 4]
```diff
@@ -9,9 +9,9 @@
     "access": "public"
   },
   "scripts": {
-    "tsc": "node node_modules/typescript/bin/tsc --project tsconfig.json",
-    "watch": "node node_modules/typescript/bin/tsc --watch --project tsconfig.json",
-    "generatePluginTypes": "rm -rf ./plugin_types && node node_modules/typescript/bin/tsc --declaration --declarationDir ./plugin_types --project tsconfig.json",
+    "tsc": "tsc --project tsconfig.json",
+    "watch": "tsc --watch --project tsconfig.json",
+    "generatePluginTypes": "rm -rf ./plugin_types && yarn run tsc --declaration --declarationDir ./plugin_types --project tsconfig.json",
```
Comment (laurent22) [inline packages/lib/package.json:14]: Need to test this.

File: packages/plugin-repo-cli/package.json [hunk 5]
```diff
@@ -3,9 +3,7 @@
   "version": "2.6.2",
   "description": "",
   "main": "index.js",
-  "bin": {
-    "plugin-repo-cli": "./index.js"
-  },
+  "bin": "./index.js",
```
Comment (laurent22) [inline packages/plugin-repo-cli/package.json:6]: I think Yarn is changing this automatically? Need to check it still works.

**Comment by laurent22** [comment]
All green now, so looks like we can try this for 2.7

**Comment by tessus** [comment]
What has to be done after the move to yarn? e.g. deleting all `node_modules` dirs?

**Comment by tessus** [comment]
We might have to update the plugin repo updater VM. I've created a snapshot of the VM so we can always rollback....

**Comment by laurent22** [comment]
I'll add a post on the forum about this. It should just be `npm run clean && yarn install`. And the plugin repo script should not be affected.

**Comment by laurent22** [comment]
Ok let's merge as it's too big to let this PR lie around. I expect there will be some problems but hopefully nothing too bad.

**Pull request merged**